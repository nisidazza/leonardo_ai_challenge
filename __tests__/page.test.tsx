import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Home from "../app/page";

describe("Home", () => {
  it("matches homepage snapshot", () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  it("renders a heading", () => {
    render(<Home />);

    expect(
      screen.getByRole("heading", {
        name: "SpaceX",
      })
    ).toBeInTheDocument();
  });

  it("renders the link to the protected page", () => {
    render(<Home />);

    const link = screen.getByRole("link", { name: "Members" });

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "/protected");
  });
});
