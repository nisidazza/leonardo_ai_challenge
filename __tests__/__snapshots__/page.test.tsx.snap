// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches homepage snapshot 1`] = `
<div>
  <div
    class="flex h-screen"
    style="background-image: url(/img.jpg); background-position: center; background-size: cover;"
  >
    <div
      class="w-screen h-screen flex flex-col justify-center items-center"
    >
      <div
        class="text-center max-w-screen-sm mb-10"
      >
        <h1
          class="text-stone-100 font-bold text-8xl"
        >
          SpaceX
        </h1>
        <p
          class="text-stone-100 mt-5 font-bold text-2xl"
        >
          Get all sorts of information about previous Space X launches!
        </p>
      </div>
      <div
        class="flex space-x-3"
      >
        <a
          class="text-stone-100 hover:text-stone-300 transition-all text-2xl"
          href="/protected"
        >
          Members
        </a>
      </div>
    </div>
  </div>
</div>
`;
